### PR TITLE
Add the GraphQL ontology definition

### PR DESCRIPTION
**Pull request by mihaimaruseac**
This should be the website's main GraphQL link. The existing one should be moved to a use case as it is an example of using GUAC via the GraphQL interface.

While here, I also fixed two typos that showed up while looking at what needs to change.

See https://github.com/guacsec/guac/pull/867

Signed-off-by: Mihai Maruseac <mihaimaruseac@google.com>

### CODE REVIEW

File: guac-graphql.md [hunk 1]
```diff
@@ -0,0 +1,823 @@
+---
+layout: page
+title: GraphQL
+permalink: /graphql/
```
Comment (jeffmendoza) [inline guac-graphql.md:4]: update title and permalink

Comment (jeffmendoza) [reply guac-graphql.md:4]: Also add `parent: GUAC use cases`

**Review by jeffmendoza — CHANGES_REQUESTED** [review]
Need to update all links `(https://github.com/guacsec/guac/blob/main/docs/GraphQL.md)` to `({{ site.baseurl }}{%link graphql.md %})`
and `({{ site.baseurl }}{%link graphql.md %})` to `({{ site.baseurl }}{%link guac-graphql.md %})`

File: graphql.md [hunk 2]
```diff
@@ -1,823 +1,765 @@
...
-permalink: /graphql/
-nav_order: 5
----
-
-# GraphQL API demo
-
-This demo introduces the GUAC GraphQL API. It covers the basic node "noun" and
-"verb" types and what they contain. It also explores the server-side `path`
-query and demonstrates a client-side search program.
+# GUAC GraphQL definitions
```
Comment (jeffmendoza) [inline graphql.md:1]: Need to add jekyll header

**Comment by mihaimaruseac** [comment]
> Need to update all links `(https://github.com/guacsec/guac/blob/main/docs/GraphQL.md)` to `({{ site.baseurl }}{%link graphql.md %})` and `({{ site.baseurl }}{%link graphql.md %})` to `({{ site.baseurl }}{%link guac-graphql.md %})`

I think so, but I haven't seen any of the existing links doing that. It seems all are handwritten.

The remaining `guacsec/guac` links link to code in the main repo (the 2 GraphQL SDLs)

**Comment by jeffmendoza** [comment]
It just merged here: https://github.com/guacsec/guac-docs/pull/43/files

**Comment by mihaimaruseac** [comment]
Oh, I need to rebase. Thanks. Doing it now

File: guac-graphql.md (hunk 1 above)
Comment (jeffmendoza) [reply guac-graphql.md:4]: still needed

**Comment by jeffmendoza** [comment]
example: https://github.com/guacsec/guac-docs/blob/main/guac-ontology-definition.md?plain=1#L14
https://github.com/guacsec/guac-docs/blob/main/guac-use-cases.md?plain=1#L14

File: guac-graphql.md (hunk 1 above)
Comment (mihaimaruseac) [reply guac-graphql.md]: Oh, missed the permalink. Done, I think

**Pull request merged**